### ISSUE TITLE
Potential bugs caused by AcceleratedOptimizer

### ISSUE THREAD

**Issue by yuxinyuan** [huggingface/accelerate#439]
Right now, `Accelerator` handles one global `GradScaler`, which will be shared by all optimizers registered in `Accelerator`, and `scaler.update` will be called in `AcceleratedOptimizer.step` method. 

If we consider the following script, the `update` method will be called twice. However, it is stated in [https://pytorch.org/docs/stable/notes/amp_examples.html#working-with-multiple-models-losses-and-optimizers](url) that the `update` method of `GradScaler` should be called only once every training step.

```
    model0, model1, optim0, optim1 = accelerator.prepare(model0, model1, optim0, optim1)

    output0 = model0(input)
    output1 = model1(input)
    loss0 = criterion(output0, target)
    loss1 = criterion(output1, target)

    accelerator.backward(loss0)
    accelerator.backward(loss1)
    # scaler.update() is called here
    optim0.step()
    # scaler.update() is called again
    optim1.step()
```

It should be noted that the above code snippet will not raise any error, so I guess it might cause some potential weird undebuggable behaviors.

**Comment by sgugger** [comment]
The gradient scaler updates are just there to adjust the rate of the multiplicative factor for the loss to avoid gradient underflow. Most of the time, nothing happens, so I don't think this would result in any bug.
